File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.cpp
class Solution {
public:
    static bool comparator(vector<int>& a, vector<int>& b) {
        return a[1] < b[1];
    }
    
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(begin(intervals), end(intervals), comparator);
        
        vector<int> prevInterval = intervals[0];
        int minRemovals = -1;
        
        for(int i = 0; i < intervals.size(); i++) {
            if(prevInterval[1] > intervals[i][0]) minRemovals++;
            
            else prevInterval = intervals[i];
        }
        
        
        return minRemovals;
    }
};

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.java
class Solution {
    static class IntervalComparator implements Comparator<int[]> {
        public int compare(int[] a, int[] b) {
            return a[1] - b[1];
        }
    }

    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, new IntervalComparator());

        int[] prevInterval = intervals[0];
        int minRemovals = 0;

        for (int i = 1; i < intervals.length; i++) {
            if (prevInterval[1] > intervals[i][0]) {
                minRemovals++;
                
            } 
            
            else prevInterval = intervals[i];
        }

        return minRemovals;
    }
}
